internal/data/metric.go

```diff
@@ -638,55 +602,47 @@ type Int64DataPoint struct {
 
 	// Override a few fields. These fields are the source of truth. Their counterparts
 	// stored in corresponding fields of "orig" are ignored.
-	pimpl *internalInt64DataPoint
-}
-
-type internalInt64DataPoint struct {
-	labels Labels
-	// True if the pimpl was initialized.
-	initialized bool
+	labels *internalLabels
 }
 
 // NewInt64DataPoint creates a new Int64DataPoint
 func NewInt64DataPoint() Int64DataPoint {
-	return Int64DataPoint{&otlpmetrics.Int64DataPoint{}, &internalInt64DataPoint{}}
+	return Int64DataPoint{&otlpmetrics.Int64DataPoint{}, &internalLabels{}}
 }
 
 // NewInt64DataPointSlice creates a slice of Int64DataPoint that are correctly initialized.
 func NewInt64DataPointSlice(len int) []Int64DataPoint {
 	// Slice for underlying orig.
 	origs := make([]otlpmetrics.Int64DataPoint, len)
 	// Slice for underlying pimpl.
-	pimpls := make([]internalInt64DataPoint, len)
+	pimpls := make([]internalLabels, len)
 	// Slice for wrappers.
 	wrappers := make([]Int64DataPoint, len)
 	for i := range origs {
 		wrappers[i].orig = &origs[i]
-		wrappers[i].pimpl = &pimpls[i]
+		wrappers[i].labels = &pimpls[i]
 	}
 	return wrappers
 }
 
 func newInt64DataPointSliceFromOrig(origs []*otlpmetrics.Int64DataPoint) []Int64DataPoint {
 	// Slice for underlying pimpl.
-	pimpls := make([]internalInt64DataPoint, len(origs))
+	pimpls := make([]internalLabels, len(origs))
 	// Slice for wrappers.
 	wrappers := make([]Int64DataPoint, len(origs))
 	for i := range origs {
 		wrappers[i].orig = origs[i]
-		wrappers[i].pimpl = &pimpls[i]
+		wrappers[i].labels = &pimpls[i]
 	}
 	return wrappers
 }
 
-func (dp Int64DataPoint) Labels() Labels {
-	dp.initInternallIfNeeded()
-	return dp.pimpl.labels
+func (dp Int64DataPoint) LabelsMap() LabelsMap {
+	return dp.labels.getLabelsMap(dp.orig.Labels)
```

Comment:
I think we need a different name for `getLabelsMap` func. Maybe `initAndGetMap`?

Reply:
Done.